### ISSUE TITLE
Flaky local e2e behavioral test in `DescendantNodes.feature`

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-development-collection#4774]
The feature NodeTraversal-DescendantNodes in `Neos.ContentRepository.BehavioralTests` seems to be flaky on todays dev-9.0.

When executing this single scenario (or all via `composer run test:behavioral:stop-on-failure`)
 
```
../../bin/behat -vvv --stop-on-failure -f progress --strict --no-interaction -c Neos.ContentRepository.BehavioralTests/Tests/Behavior/behat.yml.dist Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/NodeTraversal/DescendantNodes.feature:100
```

The assertion in line 139 https://github.com/neos/neos-development-collection/blob/dd29133357406eb823ffd0521c8d007a48e34fc8/Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/NodeTraversal/DescendantNodes.feature#L139 fails with most times with different results like:

```
        Failed asserting that two arrays are equal.
        --- Expected
        +++ Actual
        @@ @@
         Array (
        -    0 => 'a'
        -    1 => 'b'
        -    2 => 'contact'
        +    0 => 'b'
        +    1 => 'contact'
        +    2 => 'terms'
         )
```

```
        --- Expected
        +++ Actual
        @@ @@
         Array (
             0 => 'a'
        -    1 => 'b'
        -    2 => 'contact'
        +    1 => 'contact'
        +    2 => 'terms'
         )
```

rarely it succeeds locally.

But it seems that the test always passes in ci.

**Comment by mhsdesign** [comment]
Related to https://github.com/neos/neos-development-collection/issues/4769 but this should have been fixed via https://github.com/neos/neos-development-collection/pull/4770?

I checked a few states (which is kinda messy because so many repos are involved and bisect doesnt work)

```
22 sep ok
22 nov ok
27 nov ok (before sprint)
05 nov ok (after sprint)
26 nov bad (current)
```

i stopped because i found @kitsunet and @bwaidelich 's discussion

**Comment by kitsunet** [comment]
This test is broken from the start, please look at the nodes and query. None of the nodes mentioned in the result do even have the integerProperty set, how is this supposed to sort them? I am very sure the order given in the test was just the semi deterministic order it had before we changed collations, but that has nothing to do with what the test says it does IMHO. there is just no deterministic order in those nodes because they have no value for the sorting property.

tl;dr; we need to look at all test failures closely because likely they show bad tests.

As @bwaidelich and me found out the tests or rather some behaviors in the CR seemed to rely on implicit DB behavior that has changed with the collations (as in the child node sorting that basti fixed). This is another case. There just is no correct sorting here and probably in other tests either.

**Comment by mhsdesign** [comment]
As this was only a symptom this seems to be fixed with our schema adjustments. I think we can close this.

**Comment by bwaidelich** [comment]
Yes, lets re-open if it still occurs
